File: dotnet/Version3.cs
using System;
using System.Globalization;

// Version 3: Convert number to words in Spanish using native .NET library
// Usage: dotnet run --project Version3.csproj -- <number>
// Example: dotnet run --project Version3.csproj -- 10

namespace NumberConversion
{
    class Program
    {
        static void Main(string[] args)
        {
            string number = args.Length > 0 ? args[0] : "10";

            try
            {
                // Convert number to words in Spanish using NumberFormatInfo
                CultureInfo spanishCulture = new CultureInfo("es-ES");
                long num = long.Parse(number);
                
                // Use custom implementation for Spanish number to words
                string result = NumberToSpanishWords(num);
                
                Console.WriteLine(result);
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
            }
        }

        static string NumberToSpanishWords(long number)
        {
            if (number == 0) return "cero";
            
            string[] units = { "", "uno", "dos", "tres", "cuatro", "cinco", "seis", "siete", "ocho", "nueve" };
            string[] teens = { "diez", "once", "doce", "trece", "catorce", "quince", "dieciséis", "diecisiete", "dieciocho", "diecinueve" };
            string[] tens = { "", "", "veinte", "treinta", "cuarenta", "cincuenta", "sesenta", "setenta", "ochenta", "noventa" };
            string[] hundreds = { "", "ciento", "doscientos", "trescientos", "cuatrocientos", "quinientos", "seiscientos", "setecientos", "ochocientos", "novecientos" };

            if (number < 10) return units[number];
            if (number < 20) return teens[number - 10];
            if (number < 100)
            {
                int t = number / 10;
                int u = number % 10;
                return tens[t] + (u > 0 ? " y " + units[u] : "");
            }
            if (number < 1000)
            {
                int h = number / 100;
                int rest = number % 100;
                if (h == 1 && rest == 0) return "cien";
                return hundreds[h] + (rest > 0 ? " " + NumberToSpanishWords(rest) : "");
            }
            if (number < 1000000)
            {
                int thousands = (int)(number / 1000);
                int rest = (int)(number % 1000);
                if (thousands == 1) return "mil" + (rest > 0 ? " " + NumberToSpanishWords(rest) : "");
                return NumberToSpanishWords(thousands) + " mil" + (rest > 0 ? " " + NumberToSpanishWords(rest) : "");
            }
            
            return number.ToString();
        }
    }
}

File: cpp/version3.cpp
#include <iostream>
#include <string>
#include <sstream>

// Version 3: Convert number to words in Spanish using native C++ library
// Usage: g++ version3.cpp -o version3 && ./version3 <number>
// Example: ./version3 10

std::string numberToSpanishWords(long long n) {
    if (n == 0) return "cero";
    
    std::string units[] = {"", "uno", "dos", "tres", "cuatro", "cinco", "seis", "siete", "ocho", "nueve"};
    std::string teens[] = {"diez", "once", "doce", "trece", "catorce", "quince", "dieciséis", "diecisiete", "dieciocho", "diecinueve"};
    std::string tens[] = {"", "", "veinte", "treinta", "cuarenta", "cincuenta", "sesenta", "setenta", "ochenta", "noventa"};
    std::string hundreds[] = {"", "ciento", "doscientos", "trescientos", "cuatrocientos", "quinientos", "seiscientos", "setecientos", "ochocientos", "novecientos"};
    
    if (n < 10) return units[n];
    if (n < 20) return teens[n - 10];
    if (n < 100) {
        int t = n / 10;
        int u = n % 10;
        if (u == 0) return tens[t];
        return tens[t] + " y " + units[u];
    }
    if (n < 1000) {
        int h = n / 100;
        long long rest = n % 100;
        if (h == 1 && rest == 0) return "cien";
        if (rest == 0) return hundreds[h];
        return hundreds[h] + " " + numberToSpanishWords(rest);
    }
    if (n < 1000000) {
        long long thousands = n / 1000;
        long long rest = n % 1000;
        if (thousands == 1) {
            if (rest == 0) return "mil";
            return "mil " + numberToSpanishWords(rest);
        }
        if (rest == 0) return numberToSpanishWords(thousands) + " mil";
        return numberToSpanishWords(thousands) + " mil " + numberToSpanishWords(rest);
    }
    
    std::stringstream ss;
    ss << n;
    return ss.str();
}

int main(int argc, char* argv[]) {
    std::string number = argc > 1 ? argv[1] : "10";
    
    try {
        long long num = std::stoll(number);
        std::string result = numberToSpanishWords(num);
        std::cout << result << std::endl;
    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << std::endl;
    }
    
    return 0;
}
